apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -912,4 +923,25 @@ public Mono<List<WorkspaceAndResourceId>> getSpanWorkspace(@NonNull Set<UUID> sp
                         row.get("id", UUID.class))))
                 .collectList();
     }
+
+    @WithSpan
+    public Mono<Map<UUID, UUID>> getProjectIdFromSpans(@NonNull Set<UUID> spanIds) {
+
+        if (spanIds.isEmpty()) {
+            return Mono.just(Map.of());
+        }
+
+        return Mono.from(connectionFactory.create())
+                .flatMapMany(connection -> {
+
+                    var statement = connection.createStatement(SELECT_PROJECT_ID_FROM_SPANS)
+                            .bind("ids", spanIds.toArray(UUID[]::new));
+
+                    return makeFluxContextAware(bindWorkspaceIdToFlux(statement));
+                })
+                .flatMap(result -> result.map((row, rowMetadata) -> Map.entry(
+                        row.get("id", UUID.class),
+                        row.get("project_id", UUID.class))))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
Minor: this logic usually belongs to a service level, as you want to return a plain list of results. But the results of this  method are so simple (id and project_id), that this is probably ok to stay.